shared/wallets/send-form/participants/to-field.js

```diff
@@ -42,45 +56,46 @@ class ToField extends React.Component<ToFieldProps, ToFieldState> {
         this.props.onCreateNewAccount()
       } else if (element.key === 'link-existing' && this.props.onLinkAccount) {
         this.props.onLinkAccount()
-      } else if (this.props.onChangeSelectedAccount) {
-        this.setState({selectedAccount: element.props.account})
-        this.props.onChangeSelectedAccount(element.props.account.id)
+      } else {
+        this.onSelectRecipient(element.props.account)
       }
     }
   }
 
-  render() {
-    const stellarIcon = (
-      <Kb.Icon
-        type={this.props.incorrect ? 'icon-stellar-logo-grey-16' : 'icon-stellar-logo-16'}
-        style={Kb.iconCastPlatformStyles(styles.stellarIcon)}
-      />
-    )
+  showProfileOfSelectedUser = () => {
+    if (typeof this.state.selectedRecipient === 'string' && this.props.onShowProfile) {
+      this.props.onShowProfile(this.state.selectedRecipient)
+    }
+  }
 
+  render() {
     let component
 
-    if (this.props.recipientType === 'keybaseUser' && this.props.recipientUsername) {
+    // There are a few different ways the participants form can look:
+    // Case 1: A user has been set, so we display their name and avatar
+    // We can only get this case when the recipient is a stellar address or searched user, not another account.
```

Comment:
wait so, does this happen for a stellar address or does the case on L138 happen?

Reply:
It depends on if the address typed in is associated with a keybase user. If yes, then the username should replace the address [as per the mockup](https://zpl.io/aXz335K). If not, then we just show the address input (L138)